chesster.js

```diff
@@ -27,177 +30,161 @@ var BOARD_5_RATING = 16;
         return "<@"+this.getId(name)+">";
     }
 };
 var channels = {
+    list: [],
+    byId: {},
     getId: function(name){
-        return this[name].id;
+        return this.list[name].id;
     },
     getIdString: function(name){
         return "<#"+this.getId(name)+">";
     }
 };
-var channelsByID = { };
-
-
-function fuzzy_match(search_string, targets) {
-    if (!targets) { 
-        throw new Error("fuzzy_match: No targets provided");
-    }
-    var results = [];
-    targets.forEach(function(item) {
-        var distance = levenshtein.get(search_string, item);
-        results.push([distance, item]);
-    });
-    results.sort(function(a,b) {
-        return a[0] - b[0];
-    });
-    var choices = [];
-    var min_distance = results[0][0];
-    results.forEach(function(item) {
-        var distance = item[0];
-        var match = item[1];
-        if (distance == min_distance) {
-            choices.push(match);
-        }
-    });
-
-    return choices;
-}
-function get_command_and_targets(message, commands, arg_string) {
-    var command = commands[0];
-    var target = "general";
-    var channel = channelsByID[message.channel];
-    if (channel && channel.name) {
-        target = channel.name;
-    }
-    var args = arg_string;
-    if (args) {
-        args = args.split(" ");
-        if (args.length > 2) {
-            throw new Error("too many arguments to determine target / or command: ", arg_string);
-        }
-        args.forEach(function(item) {
-            var found = false;
-            commands.forEach(function(c) {
-                if (found) { return; }
-                if (levenshtein.get(c, item) < c.length / 2) {
-                    found = true;
-                    command = item;
-                }
-            });
-            if (!found) {
-                target = item;
-            }
-        });
-    }
-    command = fuzzy_match(command, commands);
-    target = fuzzy_match(target, LONEWOLF_TARGETS.concat(TEAM_TARGETS));
-    if (command.length > 1) {
-        console.log("Ambiguous command: ", command, target);
-        return {
-            command: null,
-            target: null
-        };
-    }
-    command = command[0];
-    var team_votes = 0;
-    var wolf_votes = 0;
-    target.forEach(function(item) {
-        var is_team = TEAM_TARGETS.indexOf(item) > -1;
-        var is_lonewolf = LONEWOLF_TARGETS.indexOf(item) > -1;
-        if (is_team) team_votes++;
-        if (is_lonewolf) wolf_votes++;
 
-    });
-    var retval = {
-        command: command,
-        target: null,
-    };
-    if (team_votes == wolf_votes) {
-        console.log("Ambiguous target!");
-    } else if (team_votes > wolf_votes) {
-        retval.target = "team";
-    } else if (wolf_votes > team_votes) {
-        retval.target = "lonewolf";
-    }
-    return retval;
-}
-
-controller.spawn({
-  token: process.env.token
-}).startRTM(function(err, bot) {
+function update_users(bot){
+    users.list = [];
```

Comment:
As this is async, there is a period of time during which the bot.api.users.list function below has not finished, but we have an empty users.list and functions which depend upon it will fail. I think this line should be moved into the callback below.  Basically, if we have finished AND there are no errors, then we reset the list.